packages/lib/modules/tokens/TokensProvider.tsx

```diff
@@ -110,6 +110,14 @@ export function _useTokens(
     return price.price
   }
 
+  // this also fetches the price for a bpt
+  function priceForAddress(address: string, chain: GqlChain): number {
+    const price = getPricesForChain(chain).find(price => isSameAddress(price.address, address))
+    if (!price) return 0
+
+    return price.price
+  }
```

Comment:
What's the point in this function instead of just using `priceFor`? https://github.com/balancer/frontend-monorepo/blob/ca1f61bd9c948d8c9d5f637ebed644b42099ae95/packages/lib/modules/tokens/TokensProvider.tsx#L119

Is it just that getToken doesn't work for BPT? If so, why not adapt `priceFor` to fallback to this logic if `token` isn't found?

Reply:
Tokens does not contain BPTs. I have updated `priceFor` with the fallback.